pkg/estimator/hll/hll.go

```diff
@@ -151,9 +151,10 @@ func (h *Plus) Add(v []byte) {
 
 		if uint32(len(h.tmpSet))*100 > h.m {
 			h.mergeSparse()
-			if uint32(h.sparseList.Len()) > h.m {
-				h.toNormal()
-			}
+		}
```

Comment:
It feels like this may result in h.mergeSparse() being called repeatedly.  Is it idempotent?

Reply:
Yes. we keep a small `tmpSet` and a large `sparseList`. `mergeSparse` merges the values in `tmpSet` into `sparseList` and empties `tmpSet`.

Reply:
`mergeSparse` even already has the optimisation to make it a no-op if `tmpSet` is empty